src/compiler.ts

```diff
@@ -8700,27 +8714,122 @@ export class Compiler extends DiagnosticEmitter {
       }
       let fieldInstance = <Field>member;
       let fieldType = fieldInstance.type;
-      exprs[i + 1] = this.module.store( // TODO: handle setters as well
-        fieldType.byteSize,
-        this.module.local_get(tempLocal.index, this.options.nativeSizeType),
-        this.compileExpression(values[i], fieldInstance.type, Constraints.CONV_IMPLICIT),
-        fieldType.toNativeType(),
-        fieldInstance.memoryOffset
-      );
+      exprs.push(this.module.store( // TODO: handle setters as well
+       fieldType.byteSize,
+       this.module.local_get(tempLocal.index, this.options.nativeSizeType),
+       this.compileExpression(values[i], fieldInstance.type, Constraints.CONV_IMPLICIT),
+       fieldType.toNativeType(),
+       fieldInstance.memoryOffset
+      ));
+
+      // This member is no longer omitted, so delete from our omitted fields
+      omittedClassFieldMembers.delete(member.name);
     }
     this.currentType = classReference.type.nonNullableType;
     if (hasErrors) return module.unreachable();
 
+    // Iterate through the remaining omittedClassFieldMembers.
+    if (members) {
+      for(let _keys = Map_keys(omittedClassFieldMembers), i = 0, k = _keys.length; i < k; ++i) {
+        let omittedMemberKey = _keys[i];
+        let member = assert(members.get(omittedMemberKey));
+        
+        let fieldInstance = <Field>member;
+        let fieldType = fieldInstance.type;
+
+        if (fieldType.is(TypeFlags.REFERENCE) && fieldType.classReference !== null) {
+          // TODO: Check if it is a class, with a default value (constructor with no params).
+          // TODO: Check if it can be null, and set to null
+
+          // Otherwise, error
+          this.error(
+            DiagnosticCode.Object_literal_is_missing_class_member_fields_that_must_be_defined,
+            expression.range, classReference.toString()
+          );
+          hasErrors = true;
+          continue;
+        }
+
+        switch(fieldType.kind) {
+          // i32 Types
+          case TypeKind.I8:
+          case TypeKind.I16:
+          case TypeKind.I32:
+          case TypeKind.U8:
+          case TypeKind.U16:
+          case TypeKind.U32:
+          case TypeKind.USIZE: 
+          case TypeKind.ISIZE:
+          case TypeKind.BOOL: {
+            exprs.push(this.module.store( // TODO: handle setters as well
+             fieldType.byteSize,
+             this.module.local_get(tempLocal.index, this.options.nativeSizeType),
+             this.module.i32(0),
```

Comment:
Most of this can be simplified by using 

```suggestion
             this.makeZero(fieldType),
```

so we don't need all these cases.